src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/ExpressionLambdaRewriter.cs

```diff
@@ -202,7 +202,12 @@ private BoundExpression VisitExpressionWithoutStackGuard(BoundExpression node)
                 case BoundKind.DelegateCreationExpression:
                     return VisitDelegateCreationExpression((BoundDelegateCreationExpression)node);
                 case BoundKind.FieldAccess:
-                    return VisitFieldAccess((BoundFieldAccess)node);
+                    var fieldAccess = (BoundFieldAccess)node;
+                    if (fieldAccess.FieldSymbol.IsCapturedFrame)
```

Comment:
In a given display class instance the value of captured "this", if present, should never change after being assigned. I think, native compiler did not even lift those by reference and this optimization was a result of that.

It would be difficult to observe the value of the inner frame, since there is no way it can be directly referred to it from an expression. The testcase here actually involves a nested lambda.